dom.d.ts

```diff
@@ -0,0 +1,45 @@
+type Nil = null | undefined;
+type Prim = Nil | string | number | bigint | boolean;
+
+interface Stringable {
+    toString(): string
+}
+
+type Props = Nil | Record<string, Stringable>;
+
+type Child = Prim | Node | Child[];
```

Comment:
This should be something like: `Stringable | Node | Child[]`, where `Stringable = Prim | StringableObject`. For example, passing a `URL` instance is perfectly ok.